### PR TITLE
Ability to silence DM messages from HeyFireball #14

### PR DESCRIPTION
**Pull request by JacobJerrell**
PM Functionality is fully functional for InMemoryStorage. Testing is needed for Azure, and Redis. Some housekeeping was done; commented out unused functions/methods, corrected formatting.

The bot accepts the following commands:
@[botname] setpm  -- Acts as a toggle
@[botname] setpm on
@[botname] setpm off

### CODE REVIEW

File: storage.py [hunk 1]
```diff
@@ -385,7 +395,8 @@ def create_user_entry(self, user_id: str):
         """Create new user entry and init fields."""
         self._data[user_id] = {
             self.POINTS_USED : 0,
-            self.POINTS_RECEIVED : 0
+            self.POINTS_RECEIVED : 0,
+            self.PM_PREFERENCE: 1
```
Comment (kylesykes) [inline storage.py:429]: Does this work with the table schema that exists already @MattPitlyk ?  Do we have to worry about this setting defaulting back to some value each day?

Comment (MattPitlyk) [reply storage.py:429]: This should be fine. The daily resets are for explicitly specified fields, so if we don't put the preference setting in the daily rest function, it won't be reset.

File: hey_fireball.py [hunk 2]
```diff
@@ -19,7 +19,6 @@
 BOT_ID = os.environ.get("BOT_ID")
 EMOJI = os.environ.get('EMOJI')
 POINTS = os.environ.get('POINTS')
-SELF_POINTS = os.environ.get('SELF_POINTS', "DISALLOW")
```
Comment (MattPitlyk) [inline hey_fireball.py]: Why did you remove getting this env var? It's being used in handle_command.

File: hey_fireball.py [hunk 3]
```diff
@@ -80,6 +79,9 @@ def __init__(self, msg):
                 self.target_name = self.target_id
             self.command = self._extract_command()
             self.count = self._extract_count()
+            self.setting = self._extract_setting()
+            self.setting = self._extract_setting() # Find on/off or assume toggle
```
Comment (MattPitlyk) [inline hey_fireball.py]: This line is repeated.

File: hey_fireball.py [hunk 4]
```diff
@@ -149,6 +151,29 @@ def _extract_count(self):
                     return int(self.parts[idx])
                 except ValueError:
                     pass
+
+    def _extract_setting(self):
+        if self.bot_is_first:
+            idx = 2
+            curPref = get_pm_preference(self.requestor_id)
+            try:
+                self.parts[idx]
```
Comment (MattPitlyk) [inline hey_fireball.py]: Instead of a try-except, I would check the length of self.parts to determine if which logic branch to execute. Typically, try-except is only using for flow control like this if the statement because caught does something (adds a value to a collection, gets next value in a collection, etc.), but `self.parts[idx]` does not serve a purpose other than to determine if idx=2 is value.

File: hey_fireball.py (hunk 2 above)
Comment (JacobJerrell) [reply hey_fireball.py]: I had some issues with github last night. Should be sorted out now

File: hey_fireball.py (hunk 3 above)
Comment (JacobJerrell) [reply hey_fireball.py]: A few things like this happened last night. I was trying to pick and choose what to push, and made several errors. Sorted it out, and stopped using the computer immediately

File: hey_fireball.py (hunk 4 above)
Comment (JacobJerrell) [reply hey_fireball.py]: try-except has been removed. Now we're checking if self.parts < 3
